### PR TITLE
Swipe to delete for RecyclerView example

### PR DESCRIPTION
**Pull request by beeender**
- Add parent model.
- Use RealmList for data source of RecyclerView.adapter.
- Add multiple deletion for both ListView and RecyclerView.
- Add swipe to delete for RecyclerView.
- Be prepared for adding drag & drop.

### CODE REVIEW

**Comment by beeender** [comment]
@realm/java

File: example/src/main/java/io/realm/examples/adapters/MyApplication.java [hunk 1]
```diff
@@ -30,5 +31,13 @@ public void onCreate() {
         RealmConfiguration realmConfig = new RealmConfiguration.Builder().build();
         Realm.deleteRealm(realmConfig); // Delete Realm between app restarts.
         Realm.setDefaultConfiguration(realmConfig);
+        Realm realm = Realm.getDefaultInstance();
+        realm.executeTransaction(new Realm.Transaction() {
```
Comment (cmelchior) [inline example/src/main/java/io/realm/examples/adapters/MyApplication.java]: We have 'initialData()` for this kind of stuff ;)

File: example/src/main/java/io/realm/examples/adapters/model/Counter.java [hunk 2]
```diff
@@ -15,30 +15,55 @@
...
-        this.count = count;
-    }
-
     public String getCountString() {
         return Integer.toString(count);
     }
 
-    public void increment() {
-        this.count = INTEGER_COUNTER.getAndIncrement();
+    static void create(Realm realm) {
```
Comment (cmelchior) [inline example/src/main/java/io/realm/examples/adapters/model/Counter.java]: Perhaps add comment that these needs to be called inside a write transaction?

File: example/src/main/java/io/realm/examples/adapters/ui/listview/MyListAdapter.java [hunk 3]
```diff
@@ -19,39 +19,74 @@
...
     }
 
+    private boolean inDeletionMode = false;
+    private Set<Integer> countersToDelete = new HashSet<Integer>();
+
     MyListAdapter(OrderedRealmCollection<Counter> realmResults) {
         super(realmResults);
     }
 
+    void toggleDeletionMode(boolean enabled) {
```
Comment (dalinaum) [inline example/src/main/java/io/realm/examples/adapters/ui/listview/MyListAdapter.java]: It seems not `toggle-`. How about `setDeletionMode` or `changeDeletionMode`?

File: example/src/main/java/io/realm/examples/adapters/ui/recyclerview/MyRecyclerViewAdapter.java [hunk 4]
```diff
@@ -20,20 +20,38 @@
...
+    private Set<Integer> countersToDelete = new HashSet<Integer>();
 
-    public MyRecyclerViewAdapter(RecyclerViewExampleActivity activity, OrderedRealmCollection<Counter> data) {
+    MyRecyclerViewAdapter(OrderedRealmCollection<Counter> data) {
         super(data, true);
-        this.activity = activity;
+        setHasStableIds(true);
+    }
+
+    void toggleDeletionMode(boolean enabled) {
```
Comment (dalinaum) [inline example/src/main/java/io/realm/examples/adapters/ui/recyclerview/MyRecyclerViewAdapter.java]: It also doesn't seem `toggle` method.

**Comment by dalinaum** [comment]
I just recognized that it is already merged, I will make another PR for my comments.

**Pull request merged**